src/tools/wasm-as.cpp

```diff
@@ -29,7 +30,7 @@
 
 using namespace wasm;
 
-int main(int argc, const char* argv[]) {
+int BYN_MAIN(int argc, const pchar* argv[]) {
```

Comment:
This seems a little more invasive that would imagine. 

Do we need to new `pchar` type?  Can you we have parse1 and parse2 both just take `char*`

Reply:
On windows, the unicode-aware string type is wchar_t, so at some point the codebase has to deal with wchar_t instead of char on windows. The pchar typedef here is for cases where the actual type might be wchar_t, which is the case for the unicode windows main function.

We can at least get rid of it in main by instead calling GetCommandLineW after main, on windows, ignoring the char arguments to main. I will probably make that change, since I think nobody including me, likes the BYN_MAIN macro.

As to whether we can avoid a typedef altogether that depends on whether someone can find an equivalent standard typedef for this purpose; or possibly, after removing BYN_MAIN, the rest of the code that uses pchar can be refactored to not use it.